internal/provider/cloudexport_schema.go

```diff
@@ -341,9 +341,10 @@ func resourceDataToCloudExport(d *schema.ResourceData) (*cloudexport.V202101beta
 		export.SetType(cloudexport.V202101beta1CloudExportType(type_.(string)))
 	}
 
-	if enabled, ok := d.GetOk("enabled"); ok {
-		export.SetEnabled(enabled.(bool))
-	}
+	// set the value even if "enabled" is set to type-zero value (false),
+	// so it is possible to create disabled cloudexports, and disable cloudexports that were previously enabled
+	enabled := d.Get("enabled")
+	export.SetEnabled(enabled.(bool))
```

Comment:
Can other type-zero arguments be effected? The provider should explicitly set all of them in order to avoid any hard to debug failures if the API uses a different default value than the provider assumes (i.e. uses a non type-zero value when an attribute is not present in the request). Or maybe, is there way to tell the JSON generator to always emit all attributes even those with type-zero values? Bandwidth savings achieved by not emitting them is definitely not worth the time wasted debugging problems resulting in disagreement on default values.

Reply:
Good point, I will update all the required attributes to always be set, and leave the optional ones as they are now - optional.